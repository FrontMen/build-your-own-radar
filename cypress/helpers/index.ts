export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
