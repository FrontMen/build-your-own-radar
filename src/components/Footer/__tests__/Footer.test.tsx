import React from 'react';
import { Footer } from '..';
import { withAllProviders } from 'test/helpers';

describe('Footer', () => {
  /* test to check that the mobile and larger snapshot don't change unintentionally */
  it('should matches the snapshot', () => {
    const { container} = withAllProviders(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
