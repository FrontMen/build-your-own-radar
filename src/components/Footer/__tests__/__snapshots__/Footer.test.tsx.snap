// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should matches the snapshot 1`] = `
.c0 {
  grid-area: footer;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background-color: #fff;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:481px) {
  .c0 {
    padding: 32px;
  }
}

<div>
  <div
    class="PageSlots__FooterSlot-sc-1quj5xc-2 c0"
  >
    <i>
      Â® Frontmen 2020 
    </i>
    <svg
      fill="#37383d"
      height="44"
      viewBox="-9 4 323.37 57.95"
      width="160"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M255.1 42.43V25a1.44 1.44 0 011.44-1.44h4.31a1.45 1.45 0 001.45-1.44v-8.25a1.45 1.45 0 00-1.45-1.44h-4.31A1.44 1.44 0 01255.1 11V1.44A1.45 1.45 0 00253.66 0h-10.12a1.44 1.44 0 00-1.44 1.44V11a1.45 1.45 0 01-1.45 1.44h-1.39a1.44 1.44 0 00-1.44 1.44v8.24a1.44 1.44 0 001.44 1.44h1.39A1.45 1.45 0 01242.1 25v24a8.64 8.64 0 008.62 8.63h4.38zm-162.58 0V25A1.45 1.45 0 0194 23.55h4.31a1.44 1.44 0 001.44-1.44v-8.24a1.44 1.44 0 00-1.44-1.44H94A1.45 1.45 0 0192.52 11V1.45A1.43 1.43 0 0091.08 0H81a1.44 1.44 0 00-1.44 1.44V11a1.45 1.45 0 01-1.44 1.44h-1.4a1.45 1.45 0 00-1.44 1.44v8.24a1.45 1.45 0 001.44 1.44h1.4a1.45 1.45 0 011.4 1.44v24a8.63 8.63 0 008.62 8.62h4.38zM288.9 11a23.47 23.47 0 1023.47 23.47A23.49 23.49 0 00288.9 11zm0 34.9a11.44 11.44 0 1111.44-11.43 11.45 11.45 0 01-11.44 11.45zM213.66 11a24.13 24.13 0 00-18 8A23.26 23.26 0 00190 37.68a23.75 23.75 0 0019.19 19.84 24.53 24.53 0 004.44.41c7.29 0 13.53-3 18.67-8.74l.29-.35-3-3.11A8.92 8.92 0 00219 44.18 11.38 11.38 0 01205.43 42a10.85 10.85 0 01-3-8.36 11.19 11.19 0 018.26-9.83 11.37 11.37 0 0111.38 3.36 2 2 0 001.46.66 2 2 0 001.42-.59l6.05-6.33a1.87 1.87 0 000-2.61 24.21 24.21 0 00-17.34-7.3zM160 11a23.48 23.48 0 00-4.42 46.54A23.82 23.82 0 00171.41 55l.3-.17.29.17a10.77 10.77 0 007 2.59h3.4v-42.9a2.28 2.28 0 00-2.24-2.31h-12.38A23.37 23.37 0 00160 11zm.24 35a11.54 11.54 0 1111.53-11.54A11.55 11.55 0 01160.23 46zM1.44 12.38h10.12A1.44 1.44 0 0113 13.82v43.75H8.66A8.66 8.66 0 010 48.91V13.82a1.44 1.44 0 011.44-1.44m129.83 0h-21.74a1.44 1.44 0 00-1.45 1.44v43.75h4.35a8.42 8.42 0 008.37-8.66v-15.3c0-7.18 5.29-10.26 9.93-10.26h.82a1.27 1.27 0 001.3-1.28v-8.23c.04-.84-.48-1.46-1.58-1.46zM65.92 29.25A18.26 18.26 0 0047.66 11a32.48 32.48 0 00-8 1.39H27.89a1.43 1.43 0 00-1.44 1.44V48.9a8.66 8.66 0 008.66 8.66h4.34V30.32a6.71 6.71 0 1113.41.08v18.5a8.66 8.66 0 008.66 8.66h4.34zM255.1 42.43V25a1.44 1.44 0 011.44-1.44h4.31a1.45 1.45 0 001.45-1.44v-8.25a1.45 1.45 0 00-1.45-1.44h-4.31A1.44 1.44 0 01255.1 11V1.44A1.45 1.45 0 00253.66 0h-10.12a1.44 1.44 0 00-1.44 1.44V11a1.45 1.45 0 01-1.45 1.44h-1.39a1.44 1.44 0 00-1.44 1.44v8.24a1.44 1.44 0 001.44 1.44h1.39A1.45 1.45 0 01242.1 25v24a8.64 8.64 0 008.62 8.63h4.38zm-162.58 0V25A1.45 1.45 0 0194 23.55h4.31a1.44 1.44 0 001.44-1.44v-8.24a1.44 1.44 0 00-1.44-1.44H94A1.45 1.45 0 0192.52 11V1.45A1.43 1.43 0 0091.08 0H81a1.44 1.44 0 00-1.44 1.44V11a1.45 1.45 0 01-1.44 1.44h-1.4a1.45 1.45 0 00-1.44 1.44v8.24a1.45 1.45 0 001.44 1.44h1.4a1.45 1.45 0 011.4 1.44v24a8.63 8.63 0 008.62 8.62h4.38zM288.9 11a23.47 23.47 0 1023.47 23.47A23.49 23.49 0 00288.9 11zm0 34.9a11.44 11.44 0 1111.44-11.43 11.45 11.45 0 01-11.44 11.45zM213.66 11a24.13 24.13 0 00-18 8A23.26 23.26 0 00190 37.68a23.75 23.75 0 0019.19 19.84 24.53 24.53 0 004.44.41c7.29 0 13.53-3 18.67-8.74l.29-.35-3-3.11A8.92 8.92 0 00219 44.18 11.38 11.38 0 01205.43 42a10.85 10.85 0 01-3-8.36 11.19 11.19 0 018.26-9.83 11.37 11.37 0 0111.38 3.36 2 2 0 001.46.66 2 2 0 001.42-.59l6.05-6.33a1.87 1.87 0 000-2.61 24.21 24.21 0 00-17.34-7.3zM160 11a23.48 23.48 0 00-4.42 46.54A23.82 23.82 0 00171.41 55l.3-.17.29.17a10.77 10.77 0 007 2.59h3.4v-42.9a2.28 2.28 0 00-2.24-2.31h-12.38A23.37 23.37 0 00160 11zm.24 35a11.54 11.54 0 1111.53-11.54A11.55 11.55 0 01160.23 46zM1.44 12.38h10.12A1.44 1.44 0 0113 13.82v43.75H8.66A8.66 8.66 0 010 48.91V13.82a1.44 1.44 0 011.44-1.44m129.83 0h-21.74a1.44 1.44 0 00-1.45 1.44v43.75h4.35a8.42 8.42 0 008.37-8.66v-15.3c0-7.18 5.29-10.26 9.93-10.26h.82a1.27 1.27 0 001.3-1.28v-8.23c.04-.84-.48-1.46-1.58-1.46zM65.92 29.25A18.26 18.26 0 0047.66 11a32.48 32.48 0 00-8 1.39H27.89a1.43 1.43 0 00-1.44 1.44V48.9a8.66 8.66 0 008.66 8.66h4.34V30.32a6.71 6.71 0 1113.41.08v18.5a8.66 8.66 0 008.66 8.66h4.34z"
      />
    </svg>
  </div>
</div>
`;
